front/src/modules/simulationResult/components/SpaceTimeChartWrapper/SpaceTimeChartWrapper.tsx

```diff
@@ -211,10 +213,18 @@ const SpaceTimeChartWrapper = ({
     allTrainsProjected
   );
 
-  const splitPoints = useMemo<SplitPoint[]>(() => {
-    const pathsIndex = keyBy(paths, ({ id }) => id);
+  const pathsIndex = useMemo(() => keyBy(paths, ({ id }) => id), [paths]);
```

Comment:
I know it was something that already exists but I find this name clearer 
```suggestion
  const pathsById= useMemo(() => keyBy(paths, ({ id }) => id), [paths]);
```